aspnetcore/blazor/security/index.md

```diff
@@ -0,0 +1,486 @@
+```
+
+> [!IMPORTANT]
+> Only use `[Authorize]` on `@page` components reached via the Blazor Router. Authorization is only performed as an aspect of routing and *not* for child components rendered within a page. To authorize the display of specific parts within a page, use `<AuthorizeView>` instead.
+
+You may need to add `@using Microsoft.AspNetCore.Authorization` either to the component or to the *\_Imports.razor* file in order for the component to compile.
+
+The `[Authorize]` attribute also supports role-based or policy-based authorization. For role-based authorization, use the `Roles` parameter:
+
+```cshtml
+@page "/"
+@attribute [Authorize(Roles = "admin, superuser")]
+
+<p>You can only see this if you're in the 'admin' or 'superuser' role.</p>
+```
+
+For policy-based authorization, use the `Policy` parameter:
+
+```cshtml
+@page "/"
+@attribute [Authorize(Policy = "content-editor")]
+
+<p>You can only see this if you satisfy the 'content-editor' policy.</p>
+```
+
+If neither `Roles` nor `Policy` is specified, `[Authorize]` uses the default policy, which by default is to treat:
+
+* Authenticated (signed-in) users as authorized.
+* Unauthenticated (signed-out) users as unauthorized.
+
+### Customize unauthorized content with the Router component
+
+The Router component (`<Router>`) allows the app to specify custom content if:
+
+* Content isn't found.
+* The user fails an `[Authorize]` condition applied to the component. The `[Authorize]` attribute is covered in the [[Authorize] attribute](#authorize-attribute) section.
+* Asynchronous authentication is in progress.
+
+In the default Blazor server-side project template, the *App.razor* file demonstrates how to set custom content:
+
+```cshtml
+<CascadingAuthenticationState>
+    <Router AppAssembly="typeof(Startup).Assembly">
+        <NotFoundContent>
+            <h1>Sorry</h1>
+            <p>Sorry, there's nothing at this address.</p>
+        </NotFoundContent>
+        <NotAuthorizedContent>
+            <h1>Sorry</h1>
+            <p>You're not authorized to reach this page.</p>
+            <p>You may need to log in as a different user.</p>
+        </NotAuthorizedContent>
+        <AuthorizingContent>
+            <h1>Authentication in progress</h1>
+            <p>Only visible while authentication is in progress.</p>
+        </AuthorizingContent>
+    </Router>
+</CascadingAuthenticationState>
+```
+
+The content of `<NotFoundContent>`, `<NotAuthorizedContent>`, and `<AuthorizingContent>` can include arbitrary items, such as other interactive components.
+
+If `<NotAuthorizedContent>` isn't specified, the router uses the following fallback message:
+
+```html
+Not authorized.
+```
+
+> [!NOTE]
+> **Known issue**: In ASP.NET Core 3.0 Preview 6, `<NotAuthorizedContent>` and `<AuthorizingContent>` on the Router component can't be specified in Blazor server-side apps but do work in Blazor client-side apps. This will be addressed in the Preview 7 release.
```

Comment:
This note can be removed now. Preview 7 is very close to release and does fix this.